src/WhereConditions/WhereConditionsBaseDirective.php

```diff
@@ -91,11 +91,17 @@ public function manipulateArgDefinition(
         FieldDefinitionNode &$parentField,
         ObjectTypeDefinitionNode &$parentType
     ): void {
-        if ($allowedColumns = $this->directiveArgValue('columns')) {
+        if (($allowedColumns = $this->directiveArgValue('columns')) || ($allowedColumnsEnumName = $this->directiveArgValue('columnsEnum'))) {
```

Comment:
Seems like there is some duplication between this and `@orderBy`. We can share this by adding a trait such as `GeneratesColumnsEnum`.

Reply:
I've added the trait, definitely made the code more readable and removed some duplication.